### PR TITLE
betterfox: init at 119.0

### PR DESCRIPTION
**Pull request by dani0854**
## Description of changes

user.js config for firefox

https://github.com/yokoffing/Betterfox

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by theRoboxx** [comment]
I'm new to Nixos application release schedule and like the betterfox project.
Is there an ETA when this does get merged?

File: pkgs/misc/betterfox/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  };
+
+  dontPatch = true;
+  dontConfigure = true;
+  dontBuild = true;
+
+  installPhase = ''
+    mkdir -p $out/share/betterfox
+    cp -r * $out/share/betterfox/
+  '';
```
Comment (pbsds) [inline pkgs/misc/betterfox/default.nix:24]:
```suggestion
  installPhase = ''
    runHook preInstall
    mkdir -p $out/share/betterfox
    cp -r * $out/share/betterfox/
    runHook postInstall
  '';
```

**Review by pbsds — COMMENTED** [review]
I don't really see how the user is supposed to use this declaratively? The closes option i can find is [`programs.firefox.profiles.<name>.extraConfig`](https://nix-community.github.io/home-manager/options.xhtml#opt-programs.firefox.profiles._name_.extraConfig) in home-manager, but it only accepts values known at eval-time.


>  Is there an ETA when this does get merged?

There are too few reviewers for the current influx of PRs, there are no guarantees...

File: pkgs/misc/betterfox/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  meta = with lib; {
+    description = "Firefox speed, privacy, and security: a user.js template for configuration. Your favorite browser, but better.";
```
Comment (h7x4) [inline pkgs/misc/betterfox/default.nix:27]: I think this describes what the package actually is a bit better.

```suggestion
    description = "A modern opinionated user.js template for Firefox";
```

File: pkgs/misc/betterfox/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+}:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "betterfox";
+  version = "119.0";
+
+  src = fetchFromGitHub {
+    owner = "yokoffing";
+    repo = "Betterfox";
+    rev = version;
```
Comment (h7x4) [inline pkgs/misc/betterfox/default.nix:13]: Consider using `finalAttrs`:

```suggestion
stdenvNoCC.mkDerivation (finalAttrs: {
  pname = "betterfox";
  version = "119.0";

  src = fetchFromGitHub {
    owner = "yokoffing";
    repo = "Betterfox";
    rev = finalAttrs.version;
```

**Comment by dani0854** [comment]
> The closes option i can find is [`programs.firefox.profiles.<name>.extraConfig`](https://nix-community.github.io/home-manager/options.xhtml#opt-programs.firefox.profiles._name_.extraConfig) in home-manager, but it only accepts values known at eval-time.

I used `builtins.readFile` to pass value of user.js into extraConfig. There are however a few problems using it that way, and may require a different option added to hm. Maybe NUR be a better place for this package then.

**Pull request closed without merging**